Frontend compilation fails after manual npm install

### Description of the bug

The frontend compilation of a new project downloaded from start.vaadin.com fails if one first runs `npm install`, then runs `mvn`.

The following error is displayed:
```
------------------ Starting Frontend compilation. ------------------
2022-03-08 09:13:36.429  INFO 24496 --- [onPool-worker-1] c.v.b.devserver.AbstractDevServerRunner  : Running Webpack to compile frontend resources. This may take a moment, please stand by...
2022-03-08 09:13:38.483  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [webpack-cli] Failed to load 'C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\webpack.config.js' config
2022-03-08 09:13:38.485  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [webpack-cli] Error: Cannot find module '@vaadin/build-status-plugin'
2022-03-08 09:13:38.485  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : Require stack:
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : - C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\webpack.generated.js
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : - C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\webpack.config.js
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : - C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\node_modules\webpack-cli\lib\webpack-cli.js
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : - C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\node_modules\webpack-cli\lib\bootstrap.js
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : - C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\node_modules\webpack-cli\bin\cli.js
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : - C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\node_modules\webpack-dev-server\bin\webpack-dev-server.js
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Module.require (node:internal/modules/cjs/loader:999:19)
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at require (node:internal/modules/cjs/helpers:102:18)
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Object.<anonymous> (C:\Users\Tarek Oraby\Downloads\my-app - 2022-03-08T091128.488\my-app\webpack.generated.js:16:27)
2022-03-08 09:13:38.486  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Module._compile (node:internal/modules/cjs/loader:1097:14)
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Module.load (node:internal/modules/cjs/loader:975:32)
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     at Module.require (node:internal/modules/cjs/loader:999:19) {
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :   code: 'MODULE_NOT_FOUND',
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :   requireStack: [
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     'C:\\Users\\Tarek Oraby\\Downloads\\my-app - 2022-03-08T091128.488\\my-app\\webpack.generated.js',
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     'C:\\Users\\Tarek Oraby\\Downloads\\my-app - 2022-03-08T091128.488\\my-app\\webpack.config.js',
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     'C:\\Users\\Tarek Oraby\\Downloads\\my-app - 2022-03-08T091128.488\\my-app\\node_modules\\webpack-cli\\lib\\webpack-cli.js',
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     'C:\\Users\\Tarek Oraby\\Downloads\\my-app - 2022-03-08T091128.488\\my-app\\node_modules\\webpack-cli\\lib\\bootstrap.js',
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     'C:\\Users\\Tarek Oraby\\Downloads\\my-app - 2022-03-08T091128.488\\my-app\\node_modules\\webpack-cli\\bin\\cli.js',
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :     'C:\\Users\\Tarek Oraby\\Downloads\\my-app - 2022-03-08T091128.488\\my-app\\node_modules\\webpack-dev-server\\bin\\webpack-dev-server.js'
2022-03-08 09:13:38.487  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   :   ]
2022-03-08 09:13:38.488  INFO 24496 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : }

```

### Expected behavior

No error is shown.

### Minimal reproducible example

1. Download a project from start.vaadin.com
2. from the terminal, run `npm install`
3. run `mvn`

NOTE: the error will be shown only on the first run of `mvn`. In subsequent runs, frontend compilation completes just fine.

### Versions

- Vaadin / Flow version: 23.0.0
- Java version: 17
- OS version: Windows

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
@@ -25,7 +25,6 @@ import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Collections;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
@@ -274,7 +273,7 @@ public abstract class NodeUpdater implements FallibleCommand {
 
         addDefaultObjects(packageJson);
         addVaadinDefaultsToJson(packageJson);
-        addWebpackPlugins(packageJson);
+        removeWebpackPlugins(packageJson);
 
         return packageJson;
     }
@@ -284,40 +283,26 @@ public abstract class NodeUpdater implements FallibleCommand {
         computeIfAbsent(json, DEV_DEPENDENCIES, Json::createObject);
     }
 
-    private void addWebpackPlugins(JsonObject packageJson) {
-        final List<String> plugins = WebpackPluginsUtil.getPlugins();
-
+    private void removeWebpackPlugins(JsonObject packageJson) {
         Path targetFolder = Paths.get(npmFolder.toString(), buildDir,
                 WebpackPluginsUtil.PLUGIN_TARGET);
 
-        JsonObject devDependencies;
-        if (packageJson.hasKey(DEV_DEPENDENCIES)) {
-            devDependencies = packageJson.getObject(DEV_DEPENDENCIES);
-        } else {
-            devDependencies = Json.createObject();
-            packageJson.put(DEV_DEPENDENCIES, devDependencies);
+        if (!packageJson.hasKey(DEV_DEPENDENCIES)) {
+            return;
         }
+        JsonObject devDependencies = packageJson.getObject(DEV_DEPENDENCIES);
 
         String atVaadinPrefix = "@vaadin/";
         String pluginTargetPrefix = "./"
                 + (npmFolder.toPath().relativize(targetFolder) + "/")
                         .replace('\\', '/');
-        plugins.stream().filter(plugin -> targetFolder.toFile().exists())
-                .forEach(plugin -> {
-                    String pluginTarget = pluginTargetPrefix + plugin;
-                    devDependencies.put(atVaadinPrefix + plugin, pluginTarget);
-                });
 
-        // Remove plugins previously installed but no longer needed
+        // Clean  previously installed plugins
         for (String depKey : devDependencies.keys()) {
             String depVersion = devDependencies.getString(depKey);
-            if (depKey.startsWith(atVaadinPrefix)
-                    && depVersion.startsWith(pluginTargetPrefix)) {
-                final String pluginName = depKey
-                        .substring(atVaadinPrefix.length());
-                if (!plugins.contains(pluginName)) {
-                    devDependencies.remove(depKey);
-                }
+            if (depKey.startsWith(atVaadinPrefix) && depVersion.startsWith(
+                    pluginTargetPrefix)) {
+                devDependencies.remove(depKey);
             }
         }
     }

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
@@ -297,11 +297,11 @@ public abstract class NodeUpdater implements FallibleCommand {
                 + (npmFolder.toPath().relativize(targetFolder) + "/")
                         .replace('\\', '/');
 
-        // Clean  previously installed plugins
+        // Clean previously installed plugins
         for (String depKey : devDependencies.keys()) {
             String depVersion = devDependencies.getString(depKey);
-            if (depKey.startsWith(atVaadinPrefix) && depVersion.startsWith(
-                    pluginTargetPrefix)) {
+            if (depKey.startsWith(atVaadinPrefix)
+                    && depVersion.startsWith(pluginTargetPrefix)) {
                 devDependencies.remove(depKey);
             }
         }

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdaterTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdaterTest.java
@@ -238,7 +238,7 @@ public class NodeUpdaterTest {
     }
 
     @Test
-    public void removedDisusedPlugins() throws IOException {
+    public void removedAllOldAndExistingPlugins() throws IOException {
         File packageJson = new File(npmFolder, "package.json");
         FileWriter packageJsonWriter = new FileWriter(packageJson);
         packageJsonWriter.write("{\"devDependencies\": {"
@@ -249,7 +249,7 @@ public class NodeUpdaterTest {
         JsonObject actualDevDeps = nodeUpdater.getPackageJson()
                 .getObject(NodeUpdater.DEV_DEPENDENCIES);
         Assert.assertFalse(actualDevDeps.hasKey("some-old-plugin"));
-        Assert.assertTrue(
+        Assert.assertFalse(
                 actualDevDeps.hasKey("@vaadin/application-theme-plugin"));
     }
 

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskInstallWebpackPluginsTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskInstallWebpackPluginsTest.java
@@ -94,7 +94,7 @@ public class TaskInstallWebpackPluginsTest {
     }
 
     @Test
-    public void pluginsAddedToPackageJson() throws IOException {
+    public void pluginsNotAddedToPackageJson() throws IOException {
         File resourceFolder = temporaryFolder.newFolder();
         ClassFinder finder = Mockito.mock(ClassFinder.class);
         NodeUpdater nodeUpdater = new NodeUpdater(finder,
@@ -114,16 +114,8 @@ public class TaskInstallWebpackPluginsTest {
         final JsonObject devDependencies = packageJson
                 .getObject(DEV_DEPENDENCIES);
         for (String plugin : WebpackPluginsUtil.getPlugins()) {
-            Assert.assertTrue("packageJson is missing " + plugin,
+            Assert.assertFalse("Plugin " + plugin + " added to packageJson",
                     devDependencies.hasKey("@vaadin/" + plugin));
-
-            final String pluginFolder = "./" + rootFolder.toPath()
-                    .relativize(getPluginFolder(plugin).toPath()).toString()
-                    .replace('\\', '/');
-
-            Assert.assertEquals("Plugin is pointing to wrong directory",
-                    pluginFolder,
-                    devDependencies.getString("@vaadin/" + plugin));
         }
     }
 

